R/prep_sedt_resource.R

```diff
@@ -0,0 +1,131 @@
+#' Prepare a resource for upload to the Spatial Equity Data Tool API
+#'
+#' [prep_sedt_resource()] is used by [call_sedt_api()] to convert an object or
+#' URL to a valid file path before uploading the file to the Spatial Equity Data
+#' Tool API.
+#'
+#' @param resource A URL, file path, sf or sfc object, or a data frame with
+#'   `coords` columns. See details for more information.
+#' @param coords A length 2 character vector with coordinate column names in
+#'   lon/lat order. Should correspond to the `resource_lon_column` and
+#'   `resource_lat_column` values for [call_sedt_api()].
+#' @param file File path for output file. Optional. If `NULL`, file path is set
+#'   to a temporary directory.
+#' @inheritParams rlang::args_error_context
+#' @details Valid resource inputs
+#'
+#' To call the Spatial Equity Data Tool API, [call_sedt_api()] requires a CSV or
+#' TSV input file with coordinate attribute columns. [prep_sedt_resource()] also allows users to alternatively supply:
+#'
+#' - A URL for a delimited file with coordinate columns or a spatial data file
+#' - A ArcGIS FeatureLayer or Table URL (read with [arcgislayers::arc_read()])
+#' - A file path for a spatial or tabular data file that can be read with [sf::read_sf()]
+#' - A data frame containing the named columns in `coords`
+#' - A `sf` or `sfc` object with POINT geometry that can be converted to a data frame using [sf::st_coordinates()]
+#'
+#' If resource is an `sf` object or a path to a spatial data file with POLYGON,
```

Comment:
Add text here saying:

If users plan to add use the SEDT with polygon or multi-polygon data sets, they should also use the weight column. For more details, see [here](https://ui-research.github.io/sedt_documentation/resource_datasets.html#i-have-a-dataset-of-polygons-e.g.-census-blocks.-how-can-i-use-it-with-this-tool).

Reply:
I can't edit this branch directly (since the PR is coming from the `multi-format-support` branch of this repo instead of my own branch) but I wrote up some replacement docs for the now out-dated reference to the `placement` arg that I removed over the summer:

```
#' If resource is an `sf` object or a path to a spatial data file with POLYGON,
#' MULTIPOLYGON, or any non-POINT geometry, users should pre-process the data
#' with [sf::st_centroid()] or [sf::st_point_on_surface()] and define a
#' `resource_weight` column for [call_sedt_api()] to account for differences in
#' the intersecting geometry.
```

Reply:
Done - I think we should discuss if we want to update if we add the `weight_resource_by_admin_geo()` Eli created to the package